vault/resource_mfa_okta.go

```diff
@@ -0,0 +1,143 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func mfaOktaResource() *schema.Resource {
+	return &schema.Resource{
+		Create: mfaOktaWrite,
+		Update: mfaOktaWrite,
+		Delete: mfaOktaDelete,
+		Read:   mfaOktaRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Name of the MFA method.",
+				ValidateFunc: validateNoTrailingSlash,
+			},
+			"mount_accessor": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The mount to tie this method to for use in automatic mappings. The mapping will use the Name field of Aliases associated with this mount as the username in the mapping.",
+			},
+			"username_format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A format string for mapping Identity names to MFA method names. Values to substitute should be placed in `{{}}`.",
+			},
+			"org_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the organization to be used in the Okta API.",
+			},
+			"api_token": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "Okta API key.",
+			},
+			"base_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "If set, will be used as the base domain for API requests.",
```

Comment:
nit: `If set to true....`

Reply:
@benashz this is an optional string field instead of a boolean. According to the Vault docs: `If set, will be used as the base domain for API requests. Examples are okta.com, oktapreview.com, and okta-emea.com.`